src/db.c

```diff
@@ -61,13 +61,31 @@ dict *dbIteratorNextDict(dbIterator *dbit) {
     return dbit->cur_slot >= 0 ? dbit->db->dict[dbit->cur_slot] : NULL;
 }
 
+/* Returns next entry from the multi slot db. */
+dictEntry *dbIteratorNext(dbIterator *dbit) {
+    if (!dbit->di.d) { /* No current dict, need to get a new one. */
+        dict *d = dbIteratorNextDict(dbit);
+        if (!d) return NULL;
+        dictInitSafeIterator(&dbit->di, d);
+    }
+    dictEntry *de = dictNext(&dbit->di);
+    if (!de) { /* Reached the end of the dictionary, check if there are more. */
+        dict *d = dbIteratorNextDict(dbit);
+        if (!d) return NULL;
+        dictInitSafeIterator(&dbit->di, d);
+        return dictNext(&dbit->di);
+    }
```

Comment:
Some duplicated code. Idea:

```Suggestion
    dictEntry *de = dbit->di.d ? dictNext(&dbit->di) : NULL;
    if (!de) { /* No current dict or reached the end of the dictionary. */
        dict *d = dbIteratorNextDict(dbit);
        if (!d) return NULL;
        dictInitSafeIterator(&dbit->di, d);
        return dictNext(&dbit->di);
    }
```

Reply:
I like it, looks nicer this way. Thanks.